testapi.pm

```diff
@@ -971,11 +973,11 @@ sub assert_script_run {
             # not change default timeout.
             timeout      => 90,
             fail_message => '',
-            quiet        => testapi::get_var('_QUIET_SCRIPT_CALLS')
+            quiet        => testapi::get_var('_QUIET_SCRIPT_CALLS'),
```

Comment:
I don't know what you are trying to do, but in the test you are passing `max_interval => 1000` and here I don't see `max_interval => undef` nor `max_interval => 250`

Reply:
See line 980.

Reply:
I saw line 980 and still, `%args` will contain no key `max_interval`.

Try yourself:

```bash
user@host:~$ perl missing_parameter.pl 
a: fail! 
a: fail! 
a: fail! 
b: fail! 
b: fail! 
b: do things
```

```perl
#!/usr/bin/perl
# missing_parameter.pl

sub compat_args {
    my ($def_args, $fix_keys) = splice(@_, 0, 2); 
    my %ret;
    for my $key (@{$fix_keys}) {
        $ret{$key} = shift if (scalar(@_) >= 1 && (!defined($_[0]) || !grep(/^$_[0]$/, keys(%{$def_args}))));
    }   
    carp("Odd number of arguments") unless ((@_ % 2) == 0); 
    %ret = (%{$def_args}, %ret, @_);
    map { $ret{$_} //= $def_args->{$_} } keys(%{$def_args});
    return %ret;
}

sub assert_script_run_a {
  my $cmd = shift;
  my %args = compat_args(
    {   
      timeout      => 90, 
      fail_message => '', 
      quiet        => undef
    }, ['timeout', 'fail_message'], @_);

  if ($args{max_interval}) {
    print "a: do things \n";
  } else {
    print "a: fail! \n";
  }
}

sub assert_script_run_b {
  my $cmd = shift;
  my %args = compat_args(
    {   
      timeout      => 90, 
      fail_message => '', 
      quiet        => undef,
      max_interval => undef
    }, ['timeout', 'fail_message'], @_);

  if ($args{max_interval}) {
    print "b: do things \n";
  } else {
    print "b: fail! \n";
  }
}

assert_script_run_a('true');
assert_script_run_a('true', max_interval => undef);
assert_script_run_a('true', max_interval => 250);

assert_script_run_b('true');
assert_script_run_b('true', max_interval => undef);
assert_script_run_b('true', max_interval => 250);
```

Reply:
if I set it to undef it will overwrite the default from type_string, if I set it to 250 we have duplicate code

Reply:
The code I pasted prove that `undef` doesn't overwrite.
Please, try to execute the code yourself.